### PR TITLE
Rebased Ackerman vehicle controllers

### PR DESCRIPTION
**Pull request by marcozorzi**
This is a rebased and squashed version of #7139

### CODE REVIEW

**Comment by dagar** [comment]
I think this PR has gotten away with it so far, but one thing I'll need to look at before it's mergable is the vehicle type reported by vehicle_status (commander) and any module checking it. 
Some parts of the code assume the vehicle is a FW if `vehicle_status.is_rotary_wing == false`.

**Comment by marcozorzi** [comment]
Yes, agreed. While coding I was thinking that there could be a new vehicle status, which could be wheeled, ackerman or something else but I thought that would result in refactoring other apps.

File: src/modules/gnd_pos_control/GroundRoverPositionControl.hpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+	/**
+	 * Task status
+	 *
+	 * @return	true if the mainloop is running
+	 */
+	bool		task_running() { return _task_running; }
+
+private:
+	orb_advert_t	_mavlink_log_pub{nullptr};
```
Comment (marcozorzi) [inline src/modules/gnd_pos_control/GroundRoverPositionControl.hpp]: @dagar this is unused, can we remove it? I also have a commit ready to fix the coding style, this should fix the checks issues once and for all

**Comment by TSC21** [comment]
To help on passing on travis:
https://travis-ci.org/PX4/Firmware/jobs/229043348#L416 (remove this, since it is not used)

**Comment by marcozorzi** [comment]
Thanks @TSC21, this should do it.

**Comment by TSC21** [comment]
Just picking up what I said in the other PR:
Also please add some documentation about this on https://dev.px4.io/. Create a subsection in https://dev.px4.io/en/airframes_experimental/ with the name "UGV/ROVER" or something like that (you can rename this same section to "UGV/Rover" only instead of "Boats, Submarines, Blimps, Rovers" so to be more specific), add a part list to build a RC car like yours (Traxxas Stampede VXL), an assembly guide, a general schematic of the different parts and how they are connected, the servo connections schematic, a list of mixers values and the PID tunning you used, and finally add some photos. Add everything you think it can be useful and informative.
This would help others that would like to build a UGV like yours or something similar.

**Comment by marcozorzi** [comment]
Ops, I guess this last message got lost in the final rush of my thesis. Sorry about that. I am planning on doing the guide as soon as possible, I'll keep you posted.

**Comment by dagar** [comment]
I'm still very interested in this. As soon as v1.6.0 is released let's do a final pass to get this in a mergeable state and get it in.

**Comment by marcozorzi** [comment]
As per the message above, I was finally able to create the Devguide page for this. @TSC21 let me know what you think!

**Comment by LorenzMeier** [comment]
@dagar Can you have a pass? It looked good for me.

**Comment by TSC21** [comment]
@marcozorzi thanks for the add for the Devguide! This is ok to me as is also. We can work forward on developing the modes and control to make them more independent from the fw logic.

**Comment by TSC21** [comment]
@marcozorzi after this is meged you can update https://github.com/PX4/Devguide/pull/169 so it points to the main repo instead of your repo.

**Comment by dagar** [comment]
I'll go through this again tomorrow, but it should be good to merge.

**Comment by dagar** [comment]
Merging!

**Comment by TSC21** [comment]
@marcozorzi and @dagar maybe we can schedule a call on how we can structure this to be more independent of fw code and also improve "flight" modes?

**Comment by marcozorzi** [comment]
Awesome!! I'm up for a call, thanks for the great help!

**Comment by dagar** [comment]
I'm up for it if you have specific ideas to discuss. I view this as just the start.
We can create a new UGV tag for the repo and a project board (https://github.com/PX4/Firmware/projects) if necessary.

As for the modes the first step is generalizing the FW, MC, VTOL vehicle_status flags and then in the state machine we can block modes that don't make sense for cars (ie altitude, takeoff, landing).

I'm personally interested in expanding the manual modes which we could probably get away with calling 'acro', 'stabilized', etc. Speaking for my own terrible driving I think having some kind of crude traction control or even throttle slew rate would be helpful. I suppose manual input (stick or wheel?) would simply be a yaw rate command and possibly lock on to the current heading if input is neutral?

**Comment by marcozorzi** [comment]
Sounds good to me!

**Pull request merged**